### PR TITLE
Add batch mail sending

### PR DESCRIPTION
**Pull request by Ilkrash**
https://github.com/gobuffalo/buffalo/issues/1813  
This is the proposed solution to properly send a big number of mails at the same time, with opening and closing the connection with SMTP server only one time.

### CODE REVIEW

File: mail/internal/mail/smtp.go [hunk 1]
```diff
@@ -199,7 +199,11 @@ func (d *Dialer) DialAndSend(m ...*Message) error {
 	}
 	defer s.Close()
 
-	return Send(s, m...)
+	if errs := Send(s, m...); len(errs) > 0 {
```
Comment (mclark4386) [inline mail/internal/mail/smtp.go]: as long as there are messages, the length of errs will always be > 0 (example: https://play.golang.org/p/yEQg769VNmz )

Comment (Ilkrash) [reply mail/internal/mail/smtp.go]: indeed, thank you for the response. Resolved.

**Comment by Ilkrash** [comment]
Please check the fix, tests on Windows failed due to "Test killed: ran too long (10m0s)"

File: mail/internal/mail/smtp.go [hunk 2]
```diff
@@ -199,7 +199,11 @@ func (d *Dialer) DialAndSend(m ...*Message) error {
 	}
 	defer s.Close()
 
-	return Send(s, m...)
+	if errs := Send(s, m...); errs[0] != nil {
```
Comment (mclark4386) [inline mail/internal/mail/smtp.go]: This is only checking if the first message errored. Please check all message results and send the first error found back.

**Comment by Ilkrash** [comment]
Can I do something to merge these changes? Maybe there are any suggestions to make Batch sending better?

**Pull request merged**